From 67b998473daf40c85f4a802bedccab53754c23c0 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:07:49 -0400
Subject: [powerpc] Align p_end

Message-id: <1410545655-205645-241-git-send-email-dzickus@redhat.com>
Patchwork-id: 94282
O-Subject: [RHEL7 PATCH 240/626] powerpc: Align p_end
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 286e4f90a72c0b0621dde0294af6ed4b0baddabb
Author: Anton Blanchard <anton@samba.org>
Date:   Mon Dec 23 12:19:51 2013 +1100

    powerpc: Align p_end

    p_end is an 8 byte value embedded in the text section. This means it
    is only 4 byte aligned when it should be 8 byte aligned. Fix this
    by adding an explicit alignment.

    This fixes an issue where POWER7 little endian builds with
    CONFIG_RELOCATABLE=y fail to boot.

    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/head_64.S b/arch/powerpc/kernel/head_64.S
index 2ae41ab..fad2abd 100644
--- a/arch/powerpc/kernel/head_64.S
+++ b/arch/powerpc/kernel/head_64.S
@@ -470,6 +470,7 @@ _STATIC(__after_prom_start)
  mtctr r8
  bctr
 
+.balign 8
 p_end: .llong _end - _stext
 
 4: /* Now copy the rest of the kernel up to _end */
-- 
1.7.1